core/src/components/modal/modal.tsx

```diff
@@ -75,6 +80,30 @@ export class Modal implements ComponentInterface, OverlayInterface {
    */
   @Prop() leaveAnimation?: AnimationBuilder;
 
+  /**
+   * The breakpoints to use when creating a sheet modal. Each value in the
+   * array must be a decimal between 0 and 1 where 0 indicates the modal is fully
+   * closed and 1 indicates the modal is fully open. Values are relative
+   * to the height of the modal, not the height of the screen. One of the values in this
+   * array must be the value of the `initialBreakpoint` property.
+   * For example: [0, .25, .5, 1]
+   */
+  @Prop() breakpoints?: number[];
+
+  /**
+   * A decimal value between 0 and 1 that indicates the
+   * initial point the modal will open at when creating a
+   * sheet modal. This value must also be listed in the
+   * `breakpoints` array.
+   */
+  @Prop() initialBreakpoint?: number;
```

Comment:
I think making the `initialBreakpoint` an index rather than a value would be confusing since the indices `0` and `1` could be confused for the a breakpoint values.

We could remove the requirement that `initialBreakpoint` is a value in the `breakpoints` array, but then we would have to figure out what the behavior of the initial breakpoint is: 

- Should `initialBreakpoint` be a point that users can snap to initially but can't snap to during the interaction (I.e. `initialBreakpoint` is not automatically added to the `breakpoints` array)
- Should `initialBreakpoint` be a pointer users can snap to during the interaction (I.e. `initialBreakpoint` is automatically added to the `breakpoints` array.

I would probably opt for the first option as users can get the second option by adding `initialBreakpoint` as a value in `breakpoints`.

All that said, I'm not sure there is a scenario where it makes sense to start at breakpoint `0.5`, but never let users snap back to that value. Thoughts?

Reply:
Yeah, regarding your last statement, I can't think of a reason why one would want an initial breakpoint that isn't accessible in subsequent interactions. That being said, I think the first bullet is preferable to the second bullet. 

Since you already have a console warning implemented it might be fine not to change anything, just wanted to bring this up to hear your thoughts. 🙏